x/liquidgov/keeper/icqcallbacks_mirror_proposals.go

```diff
@@ -0,0 +1,50 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+
+	"github.com/Stride-Labs/stride/v5/utils"
+	icqtypes "github.com/Stride-Labs/stride/v5/x/interchainquery/types"
+	stakeibctypes "github.com/Stride-Labs/stride/v5/x/stakeibc/types"
+)
+
+func MirrorProposalsCallback(k Keeper, ctx sdk.Context, args []byte, query icqtypes.Query) error {
+	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(query.ChainId, ICQCallbackID_MirrorProposals,
+		"Starting mirror proposals callback, QueryId: %vs, QueryType: %s, Connection: %s", query.Id, query.QueryType, query.ConnectionId))
+
+	// Confirm host exists
+	chainId := query.ChainId
+	_, found := k.stakeibcKeeper.GetHostZone(ctx, chainId)
+	if !found {
+		return sdkerrors.Wrapf(stakeibctypes.ErrHostZoneNotFound, "no registered zone for queried chain ID (%s)", chainId)
+	}
+
+	// get bytes of last entry in range query
+	lastPropBytes := sdk.InclusiveEndBytes(args)
```

Comment:
If we don't just pull the last entry, does this return a series of gov proposals?

Reply:
This is also outdated. It wouldn't pull a range, the actual query that was being sent out would just return a proposal id in the ActiveProposalQueue.